makefile

```diff
@@ -133,88 +132,8 @@ time:
-	@$(PYTHON) ./flawfinder --omittime no-ending-newline.c | \
-	  grep 'Lines analyzed' >> test-results.txt
-	@diff -u correct-results.txt test-results.txt
-
-test_002: flawfinder test.c test2.c
-	@echo 'test_002 (HTML output)'
-	@$(PYTHON) ./flawfinder --omittime --html --context test.c test2.c > test-results.html
-	@diff -u correct-results.html test-results.html
-
-test_003: flawfinder test.c test2.c
-	@echo 'test_003 (CSV output)'
-	@$(PYTHON) ./flawfinder --csv test.c test2.c > test-results.csv
-	@diff -u correct-results.csv test-results.csv
-
-test_004: flawfinder test.c
-	@echo 'test_004 (single-line)'
-	@$(PYTHON) ./flawfinder -m 5 -S -DC --quiet test.c > \
-	  test-results-004.txt
-	@diff -u correct-results-004.txt test-results-004.txt
-
-test_005: flawfinder test-diff-005.patch test-patched.c
-	@echo 'test_005 (diff)'
-	@$(PYTHON) ./flawfinder -SQDC -P test-diff-005.patch \
-	  test-patched.c > test-results-005.txt
-	@diff -u correct-results-005.txt test-results-005.txt
-
-test_006: flawfinder test.c
-	@echo 'test_006 (save/load hitlist)'
-	@$(PYTHON) ./flawfinder -S -DC --quiet \
-	  --savehitlist test-saved-hitlist-006.txt \
-	  test.c > test-junk-006.txt
-	@$(PYTHON) ./flawfinder -SQDC -m 5 \
-	  --loadhitlist test-saved-hitlist-006.txt > \
-	  test-results-006.txt
-	@diff -u correct-results-006.txt test-results-006.txt
-
-test_007: setup.py
-	@echo 'test_007 (setup.py sane)'
-	@test "`$(PYTHON) setup.py --name`" = 'flawfinder'
-	@test "`$(PYTHON) setup.py --license`" = 'GPL-2.0+'
-	@test "`$(PYTHON) setup.py --author`" = 'David A. Wheeler'
-
-test_008: flawfinder test.c
-	@echo 'test_008 (diff hitlist)'
-	@$(PYTHON) ./flawfinder -S -DC --quiet \
-	  --savehitlist test-saved-hitlist-008.txt \
-	  test.c > test-junk-008.txt
-	@$(PYTHON) ./flawfinder -S -C --quiet --omittime \
-	  --diffhitlist  test-saved-hitlist-008.txt test.c > \
-	  test-results-008.txt
-	@diff -u correct-results-008.txt test-results-008.txt
-
-# Run all tests on *one* version of Python;
-# output shows differences from expected results.
-# If everything works as expected, it just prints test numbers.
-# Set PYTHON as needed, including to ""
-test: test_001 test_002 test_003 test_004 test_005 test_006 test_007 test_008
-	@echo 'All tests pass!'
-
-# Usual check routine. Run all tests using *both* python2 and python3.
-check:
-	@echo "Testing with $(PYTHON2)"
-	@PYTHON="$(PYTHON2)" $(MAKE) test
-	@echo
-	@echo "Testing with $(PYTHON3)"
-	@PYTHON="$(PYTHON3)" $(MAKE) test
-
-# Run "make test-is-correct" if the results are as expected.
-test-is-correct: test-results.txt
-	cp -p test-results.txt correct-results.txt
-	cp -p test-results.html correct-results.html
-	cp -p test-results.csv correct-results.csv
-	cp -p test-results-004.txt correct-results-004.txt
-	cp -p test-results-005.txt correct-results-005.txt
-	cp -p test-results-006.txt correct-results-006.txt
-	cp -p test-results-008.txt correct-results-008.txt
+test:
+	$(MAKE) -C $(TESTDIR) test
```

Comment:
The POSIX standard for make does not include a "-C" option: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/make.html

I think everything is uses just POSIX make capabilities.  I'm not mortally opposed to extensions if needed, but in this case I think you can just use this instead (untested);

~~~~
cd $(TESTDIR); $(MAKE) test
~~~~

You should also pass through "make check" (unless that's already done elsewhere).

Reply:
Narg, in fact, I did not check if it is POSIX compliant, I will replace it with your proposed solution and also passthrough the `check` target.

Reply:
I will also reintroduce the `test-is-correct` target.